install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -10,6 +10,7 @@
 
 {{- $kubeProxyReplacement := (coalesce .Values.kubeProxyReplacement "false") -}}
 {{- $envoyDS := eq (include "envoyDaemonSetEnabled" .) "true" -}}
+{{- $buildDaemonConfig := not (eq .Values.daemon.configSources "config-map:cilium-config") -}}
```

Comment:
Wouldn't we also want to omit the initContainer in the case that we're only using a config-map, but it's being configured to be something other than `cilium-config`, for ex: `--values daemon.configSources=config-map:some-config`?

Reply:
Sure, it's easy, and would be consistent with what the build-config init container does. But the configmap name `cilium-config` is hard-coded everywhere else in the chart templates (cilium-operator, cilium-preflight, clustermesh-apiserver), so I am not sure what that changing just the daemon's configmap would do.